### PR TITLE
Clarify Hash literal syntax. Expand on other things.

### PR DESCRIPTION
**Pull request by awwaiid**

### CODE REVIEW

File: doc/Language/syntax.pod [hunk 1]
```diff
@@ -395,6 +406,23 @@ You can also assign subroutines to variables.
...
+Subroutines can be declared with multiple signatures.
+
+    multi sub foo() { say "Hello!" }
+    multi sub foo($name) { say "Hello $name!" }
+
+Inside of a class, you can also declare multi-dispatch methods.
+
+    multi method greet { }
+    multi method greet(Str $name) { }
+
```
Comment (LLFourn) [inline doc/Language/syntax.pod]: somehow link to: http://docs.perl6.org/language/functions#Multi-dispatch "for more info" or you could just L<> the heading.

I plan to document `proto` there soon too.

**Comment by LLFourn** [comment]
Thanks again!

File: doc/Language/syntax.pod (hunk 1 above)
Comment (awwaiid) [inline doc/Language/syntax.pod]: I suppose I should learn what `proto` is :)

Comment (LLFourn) [inline doc/Language/syntax.pod]: Hmm I just realised that `L<>` in headings looks like a bad idea. As you can see it's in the `Int literals` heading, which seems to goof the table of contents link.

Probably best to purge them from headings for now.

**Pull request merged**